### PR TITLE
fix: additional tweaks for did:web and other methods as public DIDs

### PR DESCRIPTION
**Pull request by dbluhm**
This PR adjusts the request schema for the set public DID endpoint to permit `GENERIC_DID`s. It seems that the work to support a generic DID in this endpoint was already done and just the request schema was missed by mistake (or perhaps it's a merge artifact :man_shrugging:).

Additionally, this PR adjusts the behavior of DID Exchange requests so that if we're using a public DID, no DID Doc attachment is included in the request. This enables us to form connections using a did:web or other DID that is resolvable without worrying about ACA-Py's DID Doc creation process not being quite up to snuff on multi-method support.

And some minor type hint corrections.

### CODE REVIEW

**Comment by dbluhm** [comment]
I should add some unit tests for this. Marking as draft until I get that taken care of.

**Review by usingtechnology — APPROVED** [review]
looks good, test is nice to have.
very minor comment to which I have no feelings either way - update at your discretion.

File: aries_cloudagent/protocols/didexchange/v1_0/manager.py [hunk 1]
```diff
@@ -245,11 +246,12 @@ async def create_request_implicit(
...
-        my_endpoint: str = None,
-        mediation_id: str = None,
-        goal_code: str = None,
-        goal: str = None,
+        my_label: Optional[str] = None,
+        my_endpoint: Optional[str] = None,
+        mediation_id: Optional[str] = None,
+        goal_code: Optional[str] = None,
+        goal: Optional[str] = None,
+        public_did: bool = False,
```
[the remark this answers is not in the document]
Comment (dbluhm) [reply aries_cloudagent/protocols/didexchange/v1_0/manager.py]: Thanks for the feedback! Addressed in most recent commit.

**Review by usingtechnology — APPROVED** [review]
Good stuff.

**Pull request merged**